install/charts/templates/delete-clean-up.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2019 SAP SE or an SAP affiliate company. All rights reserved.
+# This file is licensed under the Apache Software License, v. 2 except as
+# noted otherwise in the LICENSE file.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Values.metadata.name }}-delete
+  labels:
+    app: {{ .Values.metadata.labelApp }}
+  namespace: {{ .Values.metadata.namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
```

Comment:
`post-delete` is only possible, if we use another service account that persists after karydia is deleted, as otherwise the needed permissions are not available after the deletion of the karydia service account.

Moreover, as one wants to rewind the installation process, it makes sense to first delete the webhooks (as they were installed last). However, one would like to delete the CRDs last (as they were installed first). One could do both using a `pre-delete` and a `post-delete` hook in two separate pods. This might be a too big overhead for such a simple task.

Reply:
I would not (re-)use the karydia ServiceAccount for its installation / deletion. A more detailed explanation: https://github.com/karydia/karydia/pull/199#discussion_r316531951

My thought is that if you use a `pre-delete` hook you delete components required by karydia to run correctly. If the deletion breaks, karydia could still be there but wouldn't work anymore. With a `post-delete` hook: If the deletion breaks, only some orphaned dependencies could remain but it wouldn't break karydia.

Reply:
Changed to `post-delete` while using a separate serviceAccount.